src/main/java/net/contargo/types/telephony/formatting/PhoneNumberFormatter.java

```diff
@@ -216,26 +225,29 @@ private AreaCodeAndConnectionNumber getAreaCodeAndConnectionNumberFromFixedNumbe
             throw new PhoneNumberFormattingException(String.format("Could not extract anything from input number: %s",
                     preFormattedNumber));
         } else if (splittedPreformattedNumber.length <= 2) { // two elements -> take the second element as connection
```

Comment:
Not sure if that really applies to reality - but what if there are more than 3 elements e.g. more spaces in the number. Imagine a number within a company where you could have [countrycode] [city code] [connection number] [extension]. Normally that extension would be seperated by a '-'. But since the task of this lib is formatting malformed numbers we should expect something like this.

To cut a long story short. I'd have code that takes all remaining elements and hand them over as VarArgs to the NewNationalNumber constructor.

Reply:
OK. With some support I read the code again and it handles exactly the described case. ;)